### DISCUSSION TITLE
Long term limits

### DISCUSSION THREAD

**Discussion by programmer1855** [fattureincloud/api#275] [Q&A] [answered]
Buongiorno,
vorrei avere un chiarimento sui limiti a lungo termine, in particolare quelli mensili.

Dal momento che il numero di chiamate API è limitato, sto tenendo traccia di quante chiamate vengono consumate.

Ho realizzato un'app pubblica a cui si collegano diversi clienti, ma non sto vedendo un calo significativo delle chiamate API disponibili nonostante l'app venga utilizzata spesso.

Vi chiedo quindi se i limiti di utilizzo di 40000 chiamate si applicano all'app o ai singoli clienti che utilizzano l'app?

Ogni cliente che usa l'app pubblica ha 40000 chiamate mensili a disposizione?

Grazie

**Comment by valmoz** [comment, marked as answer]
Buongiorno @programmer1855 
come da [documentazione](https://developers.fattureincloud.it/docs/basics/limits-and-quotas), le quote si comportano in maniera differente per app private/pubbliche.

- Per le app pubbliche, ogni azienda ha 40K richieste su quell'app (si valuta sulla coppia company-app)
- Per le app private, ogni azienda ha 40K richieste condivise tra tutte le app private (si calcola sulla sola company, indipendentemente dall'app privata utilizzata)

Quindi immaginiamo di avere due aziende, AAA e BBB, due app private 1111 e 2222 e due app pubbliche 9999 e 0000.
Le quote saranno così distribuite:
AAA avrà le seguenti quote:
- 40K per 9999
- 40K per 0000
- 40K condivise tra 1111 e 2222

BBB avrà le medesime quote:
- 40K per 9999
- 40K per 0000
- 40K condivise tra 1111 e 2222

I consumi di AAA non influiscono su quelli di BBB; quindi se AAA effettuerà 30K chiamate da 9999 la sua quota residua su tale app scenderà a 10K, mentre per BBB la quota resterà inalterata a 40K.

Le quote di AAA e BBB sono distinte: qualora AAA dovesse raggiungere il limite per l'app 9999, BBB potrebbe comunque continuare a consumare le API per 9999 in quanto le sue quote sono separate; anche se le sue quote per 9999 sono consumate, AAA potrebbe comunque continuare ad usare 0000, perché le quote si riferiscono alla coppia company-app e quindi sono separate.

Per le app private la questione è diversa. Supponiamo che AAA faccia 40K richieste con l'app 1111. In questo caso, le quote risulterebbero esaurite sia per 1111 che per 2222, in quanto la quota è condivisa tra tutte le app private. AAA potrebbe continuare ad usare le app pubbliche (a patto di avere quote su tali app), perché le app pubbliche hanno una quota separata, e BBB potrebbe comunque continuare ad usare le app 1111 e 2222, in quanto è la quota delle app private di AAA ad essere esaurita.

Spero di esser stato chiaro.

Grazie
Mauro
